docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,303 @@
+
+## Desired user experience for proposal 1
+
+For review convenience, this doc shows the output in most of the sample ORAS commands with prettified JSON format.
+
+### oras pull 
+
+Pull an artifact and display its metadata as formatted JSON in standard output. The following fields should be formatted in a JSON output:
+
+- `ref`: full artifact reference by digest, e.g, `$REGISTRY/$REPO@$DIGEST`
+-  `files`: a list of downloaded files
+    -  `path`: the absolute file path of the pulled file (layer)
+    -  `ref`: full reference by digest of the pulled file (layer)
+    -  `mediaType`: media type of the pulled file (layer) 
+    -  `digest`: digest of the pulled file (layer) 
+    -  `size`: file size in bytes
+
+Pull a single file and show the its descriptor data including `path` and `ref` as pretty JSON in standard output:
+
+```bash
+oras pull $REGISTRY/$REPO:$TAG --format '{{toPrettyJson .}}'
+```
+
+```json
+{
+    "ref": "$REGISTRY/$REPO@$DIGEST",
+    "files" : [
+            "path":"/home/user/path1/",
+            "ref": "$REGISTRY/$REPO@$layer0_digest",
+            {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar",
+            "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+            "size": 12
+            }
+        }
+    ]
+}
+```
+
+Pull an artifact and display its descriptor as raw JSON in standard output.
+
+```bash
+oras pull $REGISTRY/$REPO:$TAG --format json
+```
+
+```
+{"ref":"$REGISTRY/$REPO@$DIGEST","files":[{"path":"/home/user/path1/","ref":"$REGISTRY/$REPO@$layer0_digest","mediaType":"application/vnd.oci.image.manifest.v1+json","digest":"sha256:42e2c5e85dd5a21dd516dd6f5a043db9ae549b8f464b049d165fc5765ebb4cad","size":591}]}
+```
+
+Pull multiple files and show their descriptor data including `path` and `ref` as pretty JSON in standard output.
+
+```bash
+oras pull $REGISTRY/$REPO:$TAG --format '{{toPrettyJson .}}'
+```
+
+```json
+{
+    "ref": "$REGISTRY/$REPO@$DIGEST",
+    "files" : [
+        {
+            "path":"path1/artifact1.json",
+            "ref": "$REGISTRY/$REPO:$layer0_digest",
+            "mediaType": "application/vnd.oci.image.layer.v1.tar",
+            "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+            "size": 12,
+        },
+        {
+            "path":"path2/artifact2.json",
+            "ref": "$REGISTRY/$REPO:$layer1_digest",
+            "mediaType": "application/vnd.oci.image.layer.v1.tar",
+            "digest": "sha256:4add5a911ba64df27458ed8229da804a26d2a84f4b8b650937ec8f73cd8be2c7",
+            "size": 12,
+        }
+        }
+    ]
+}
+```
+
+### oras attach
+
+Attach two files to an image and show the descriptor of the attached files in JSON format.
+
+```bash
+oras attach $REGISTRY/$REPO:$TAG --artifact-type example/sbom sbom.spdx --artifact-type example/vul-scan vul-report.json  --format '{{toPrettyJson .}}'
+```
+
```

Comment:
Why do we need to emit minified or condensed json in any of these. jq and other tools always prettify. I would vote for removing the option of toPrettyJson and and if we need minified json or raw introduce another option when needed.

Reply:
If we output prettified JSON metadata by default, we don't need to introduce another option to minify it. It's already supported natively and user can use `--format '{{toRawJson .}}'`.